packages/common/src/external/lecternClient.ts

```diff
@@ -13,14 +13,16 @@ const client = (schemaServiceUrl: string, logger: Logger) => {
 		 * @returns A Dictionary found
 		 */
 		async fetchDictionaryByVersion(name: string, version: string): Promise<dictionaryEntities.SchemasDictionary> {
+			let newSchema;
 			try {
-				const newSchema = await dictionaryRestClient.fetchSchema(schemaServiceUrl, name, version);
-				if (!newSchema) throw new BadRequest(`Schema with name '${name}' and version '${version}' not found`);
-				return newSchema;
+				newSchema = await dictionaryRestClient.fetchSchema(schemaServiceUrl, name, version);
 			} catch (error) {
 				logger.error(LOG_MODULE, `Error Fetching dictionary from lectern`, error);
 				throw new ServiceUnavailable();
 			}
+
+			if (!newSchema) throw new BadRequest(`Schema with name '${name}' and version '${version}' not found`);
+			return newSchema;
```

Comment:
If you move this inside the try block then we don't need the mutable `newSchema` variable.

Reply:
Left outside of the try block as we want to identify 2 types of error. 
1. The try/catch block throws a ServiceUnavailable when Lectern service is down. 
2. Throw a BadRequest in case dictionary name/version doesn't exist in Lectern.

Reply:
Also since this function is throwing errors I checked for uses of it to make sure they are being properly wrapped in try/catch blocks... there is a usage in register dictionary without a try/catch, this means we could be throwing a `ServiceUnavailable` error from our controller. Do we have a clear response for this type in our error handler?